src/runtime/virtcontainers/hypervisor.go

```diff
@@ -564,6 +564,11 @@ func (conf *HypervisorConfig) Valid() error {
 		conf.DefaultMaxVCPUs = defaultMaxVCPUs
 	}
 
+	if conf.ConfidentialGuest && conf.NumVCPUs != conf.DefaultMaxVCPUs {
+		hvLogger.Warnf("Confidential guests do not support hotplugging of vCPUs. Setting DefaultMaxVCPUs to NumVCPUs (%d)", conf.NumVCPUs)
+		conf.DefaultMaxVCPUs = conf.NumVCPUs
```

Comment:
I'm not sure this is the best policy here -- this isn't really confidential specific -- other VMM configurations also have this limitation.

Reply:
So, from my point either we do this here or we let it error out on the VMM side.
See the answer below, but AFAIU this is a "limitation" of Confidential Computing and not related to the VMM itself.

I am not sure if I follow you on "this isn't confidential specific", mind to expand?

Reply:
Other VMMs and architectures also don't support hot plug (see Firecracker, see baseline aarch64).

If we want to setup a sandbox sizing policy when hot plug isn't available, then we should make it generic -- this isn't a confidential computing specific problem.

Reply:
or, perhaps this is me err; All we're doing is setting default MAX equal to whatever is set for the machine.

Curious -- does CLH and or QEMU fail to boot of max cpu is greater than number of CPU, then?

Reply:
CLH for sure.  I think QEMU does as well.

Reply:
/cc @devimc

Reply:
>  does CLH and or QEMU fail to boot of max cpu is greater than number of CPU, then?

yes